recipe/meta.yaml

```
@@ -1,5 +1,5 @@
 {% set version = "3.1.0a1" %}
```

I'd prefer that something which isn't a tagged commit doesn't pretend to be one. We could either just go back to `.dev1`, or if that messes up the order, we could do something like
```suggestion
{% set version = "3.1.0a1.dev0" %}
```

---

Your comment on gh-149 was: _You should probably keep the version as `3.1.0a1` though (and just increase the build number), as I think that conda will otherwise keep prioritising `a1` builds over `dev0` ones._

Unless that's not true, I think it should indeed be `3.1.0a1`.

---

Yeah, I had said:
> You should probably keep the version as `3.1.0a1` though (and just increase the build number), as I think that conda will otherwise keep prioritising `a1` builds over `dev0` ones.

But that was being concerned about conda actually pulling in an older `.0a1` over a newer `.0dev0` build. That concern is still relevant (we should pull in the newest builds by default), but I'd like to sharpen what I said before to _also_ avoid dev builds masquerading as a given tagged version. Let's try what happens with my suggestion above.

---

This doesn't work, it should have been `3.1.0a2.dev0` (towards `a2`, not `a1`), as you can see from `mamba search -c conda-forge/label/cython_dev` or installing from the dev label.

Shall I open a new PR?

---

Fixed in https://github.com/conda-forge/cython-feedstock/commit/bccdf11b683534ee0c6606129e35ef1e132e26df

---

thanks!

---

Still not working. `conda search` sorts versions in the desired order now, but I can't find any way to get `conda` or `mamba` to install `3.1.0a2.dev0`. E.g.:
```
$ conda install conda-forge/label/cython_dev::cython
```
doesn't work.

Can you find a way?

---

```
mamba install -c conda-forge/label/cython_dev cython=3.1.0a2.dev0
```
works for me... 🤔

---

Hmm, that works as does this:
```
$ mamba install -c conda-forge/label/cython_dev cython
```

But not in an environment that already has `python-freethreading`. And if you try both at once it doesn't work even with an exact pin (while `=3.1.0a1` does work):
```
$ mamba install -c conda-forge/label/cython_dev cython=3.1.0a2.dev0 python-freethreading
...
Encountered problems while solving:
  - package python-freethreading-3.13.0-h92d6c8b_0 requires python_abi * *_cp313t, but none of the providers can be installed
```

So something is still wrong with either this particular build or with using a `.dev0` suffix.

---

we don't have 3.13t builds for `cython`, which is reflected in 
```
    The following packages are incompatible
    ├─ cython =3.1.0a2.0dev0 * is installable with the potential options
    │  ├─ cython 3.1.0a2.dev0 would require
    │  │  └─ python_abi =3.13 *_cp313, which can be installed;
    │  ├─ cython 3.1.0a2.dev0 would require
    │  │  └─ python >=3.10,<3.11.0a0 *, which can be installed;
     [etc.; no variant for 3.13t]
```
That's because we lost the `noarch: True` builds in https://github.com/conda-forge/cython-feedstock/commit/0f0fef38d262d11904316fcfad9c0ac66d48ae1f - something I overlooked while reviewing this PR on a phone...

---

🤔 I hadn't even considered checking the output of `conda-smithy rerender`, I thought that was always safe.

---

Usually it is; sometimes changes in conda-build (which determines which variables are considered active) or conda-smithy can lead to unwanted changes though...

---

Thanks for the fix! - I confirm it's working now.